src/Core/src/Handlers/Window/WindowHandler.iOS.cs

```diff
@@ -15,16 +15,32 @@ protected override void ConnectHandler(UIWindow platformView)
 		public static void MapTitle(IWindowHandler handler, IWindow window) =>
 			handler.PlatformView.UpdateTitle(window);
 
+#if MACCATALYST
+		internal static void MapTitleBar(IWindowHandler handler, IWindow window)
+		{
+			if (handler.PlatformView.RootViewController is WindowViewController controller && handler?.MauiContext is IMauiContext mauiContext)
+			{
+				controller.SetUpTitleBar(window, mauiContext, false);
+			}
+		}
+#endif
+
 		public static void MapContent(IWindowHandler handler, IWindow window)
 		{
 			_ = handler.MauiContext ?? throw new InvalidOperationException($"{nameof(MauiContext)} should have been set by base class.");
 
 			var nativeContent = window.Content.ToUIViewController(handler.MauiContext);
 
+			var rootViewController = handler.PlatformView.RootViewController;
+
+#if MACCATALYST
+			// If we are in Catalyst, use the WindowViewController so we can handle the TitleBar
+			handler.PlatformView.RootViewController = new WindowViewController(nativeContent, window, handler.MauiContext);
+#else
 			handler.PlatformView.RootViewController = nativeContent;
+#endif
 
-			if (window.VisualDiagnosticsOverlay != null)
-				window.VisualDiagnosticsOverlay.Initialize();
+			window.VisualDiagnosticsOverlay?.Initialize();
```

Comment:
No longer requires to check:

`if (window.VisualDiagnosticsOverlay != null)`

?

Reply:
With this format, if window.VisualDiagnosticsOverlay is null, it won't run the Initialize method :)